Sources/RabbitMq/Publisher/RetryingPublisher.swift

```diff
@@ -2,31 +2,38 @@ import AMQPClient
 import Logging
 import NIOCore
 
-struct RetryingPublisher: Sendable {
+struct RetryingPublisher: ConnectionLoggable {
     let connection: Connection
     let configuration: PublisherConfiguration
-    let logger: Logger
     let retryInterval: Duration
 
+    var logger: Logger {
+        get async {
+            await connection.logger.withMetadata(["exchangeName": .string(configuration.exchangeName)])
```

Comment:
This creates a new logger with metadata every time `logger` is used. This is done even when the log is not actually created due to the current log level.